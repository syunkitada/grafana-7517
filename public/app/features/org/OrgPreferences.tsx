import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Label } from '../../core/components/Label/Label';
import SimplePicker from '../../core/components/Picker/SimplePicker';
import { DashboardSearchHit, OrganizationPreferences } from 'app/types';
import {
  setOrganizationHomeDashboard,
  setOrganizationTheme,
  setOrganizationTimezone,
  updateOrganizationPreferences,
} from './state/actions';

export interface Props {
  preferences: OrganizationPreferences;
  starredDashboards: DashboardSearchHit[];
  setOrganizationHomeDashboard: typeof setOrganizationHomeDashboard;
  setOrganizationTheme: typeof setOrganizationTheme;
  setOrganizationTimezone: typeof setOrganizationTimezone;
  updateOrganizationPreferences: typeof updateOrganizationPreferences;
}

const themes = [{ value: '', text: 'Default' }, { value: 'dark', text: 'Dark' }, { value: 'light', text: 'Light' }];

const timezones = [
  { value: '', text: 'Default' },
  { value: 'browser', text: 'Local browser time' },
  { value: 'utc', text: 'UTC' },
];

export class OrgPreferences extends PureComponent<Props> {
  onSubmitForm = event => {
    event.preventDefault();
    this.props.updateOrganizationPreferences();
  };

  render() {
    const {
      preferences,
      starredDashboards,
      setOrganizationHomeDashboard,
      setOrganizationTimezone,
      setOrganizationTheme,
    } = this.props;

    starredDashboards.unshift({ id: 0, title: 'Default', tags: [], type: '', uid: '', uri: '', url: '' });

    return (
      <form className="section gf-form-group" onSubmit={this.onSubmitForm}>
        <h3 className="page-heading">Preferences</h3>
        <div className="gf-form">
          <span className="gf-form-label width-11">UI Theme</span>
          <SimplePicker
            defaultValue={themes.find(theme => theme.value === preferences.theme)}
            options={themes}
            getOptionValue={i => i.value}
            getOptionLabel={i => i.text}
            onSelected={theme => setOrganizationTheme(theme.value)}
            width={20}
          />
        </div>
        <div className="gf-form">
          <Label
            width={11}
            tooltip="Not finding dashboard you want? Star it first, then it should appear in this select box."
          >
            Home Dashboard
          </Label>
          <SimplePicker
            defaultValue={starredDashboards.find(dashboard => dashboard.id === preferences.homeDashboardId)}
            getOptionValue={i => i.id}
            getOptionLabel={i => i.title}
            onSelected={(dashboard: DashboardSearchHit) => setOrganizationHomeDashboard(dashboard.id)}
            options={starredDashboards}
            placeholder="Chose default dashboard"
            width={20}
          />
        </div>
        <div className="gf-form">
          <label className="gf-form-label width-11">Timezone</label>
          <SimplePicker
            defaultValue={timezones.find(timezone => timezone.value === preferences.timezone)}
            getOptionValue={i => i.value}
            getOptionLabel={i => i.text}
            onSelected={timezone => setOrganizationTimezone(timezone.value)}
            options={timezones}
            width={20}
          />
        </div>
        <div className="gf-form-button-row">
          <button type="submit" className="btn btn-success">
            Save
          </button>
        </div>
      </form>
    );
  }
}

function mapStateToProps(state) {
  return {
    preferences: state.organization.preferences,
    starredDashboards: state.organization.starredDashboards,
  };
}

const mapDispatchToProps = {
  setOrganizationHomeDashboard,
  setOrganizationTimezone,
  setOrganizationTheme,
  updateOrganizationPreferences,
};

export default connect(mapStateToProps, mapDispatchToProps)(OrgPreferences);
