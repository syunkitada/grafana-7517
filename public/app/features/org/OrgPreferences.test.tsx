import React from 'react';
import { shallow } from 'enzyme';
import { OrgPreferences, Props } from './OrgPreferences';

const setup = () => {
  const props: Props = {
    preferences: {
      homeDashboardId: 1,
      timezone: 'UTC',
      theme: 'Default',
    },
    starredDashboards: [{ id: 1, title: 'Standard dashboard', url: '', uri: '', uid: '', type: '', tags: [] }],
    setOrganizationTimezone: jest.fn(),
    setOrganizationTheme: jest.fn(),
    setOrganizationHomeDashboard: jest.fn(),
    updateOrganizationPreferences: jest.fn(),
  };

  return shallow(<OrgPreferences {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
