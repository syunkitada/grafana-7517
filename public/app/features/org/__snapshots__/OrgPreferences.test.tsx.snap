// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<form
  className="section gf-form-group"
  onSubmit={[Function]}
>
  <h3
    className="page-heading"
  >
    Preferences
  </h3>
  <div
    className="gf-form"
  >
    <span
      className="gf-form-label width-11"
    >
      UI Theme
    </span>
    <SimplePicker
      getOptionLabel={[Function]}
      getOptionValue={[Function]}
      onSelected={[Function]}
      options={
        Array [
          Object {
            "text": "Default",
            "value": "",
          },
          Object {
            "text": "Dark",
            "value": "dark",
          },
          Object {
            "text": "Light",
            "value": "light",
          },
        ]
      }
      width={20}
    />
  </div>
  <div
    className="gf-form"
  >
    <span
      className="gf-form-label width-11"
    >
      Home Dashboard
      <class_1
        className="gf-form-help-icon gf-form-help-icon--right-normal"
        content="Not finding dashboard you want? Star it first, then it should appear in this select box."
        placement="right"
      >
        <i
          className="fa fa-info-circle"
        />
      </class_1>
    </span>
    <SimplePicker
      defaultValue={
        Object {
          "id": 1,
          "name": "Standard dashboard",
        }
      }
      getOptionLabel={[Function]}
      getOptionValue={[Function]}
      onSelected={[Function]}
      options={
        Array [
          Object {
            "id": 1,
            "name": "Standard dashboard",
          },
        ]
      }
      placeholder="Chose default dashboard"
      width={20}
    />
  </div>
  <div
    className="gf-form"
  >
    <label
      className="gf-form-label width-11"
    >
      Timezone
    </label>
    <SimplePicker
      getOptionLabel={[Function]}
      getOptionValue={[Function]}
      onSelected={[Function]}
      options={
        Array [
          Object {
            "text": "Default",
            "value": "",
          },
          Object {
            "text": "Local browser time",
            "value": "browser",
          },
          Object {
            "text": "UTC",
            "value": "utc",
          },
        ]
      }
      width={20}
    />
  </div>
  <div
    className="gf-form-button-row"
  >
    <button
      className="btn btn-success"
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
